### PR TITLE
MAINT: Ressources → Resources (and internal name childs)

### LINKED ISSUE #2548: Migrate public APIs containing typos
We currently have some public APIs with typos which should be migrated accordingly:

* `pypdf.constants.Ressources` to `pydf.constants.Resources`
* `pypdf.generic._data_structures.Destination.childs` to `pypdf.generic._data_structures.Destination.children` (already has a TODO)

### PR DESCRIPTION
**Pull request by pubpub-zz**
closes #2548

### CODE REVIEW

**Comment by pubpub-zz** [comment]
I've not been able to change childs into children : the Destination(Object) derivates from TreeObjects which has `children` member. any ideas

**Comment by stefan6419846** [comment]
> the Destination(Object) derivates from TreeObjects which has children member. any ideas

Does the `Destination` need both attributes or do we only need one - id est, what is the difference between `children` and `childs` at the moment?

**Comment by pubpub-zz** [comment]
actually the field is for pypdf internal use only. I've fixed the name

File: pypdf/generic/_data_structures.py [hunk 1]
```diff
@@ -1416,7 +1416,7 @@ class Destination(TreeObject):
     node: Optional[
         DictionaryObject
     ] = None  # node provide access to the original Object
-    childs: List[
+    _filtered_children: List[
         Any
     ] = []  # used in PdfWriter - TODO: should be children  # noqa: RUF012
```
Comment (stefan6419846) [inline pypdf/generic/_data_structures.py:1421]: We can probably remove the TODO comment here.

Another thing: Do we really need this at class level or can be move this and the initialization to the `__init__` method? The *RUF012* violation we silence here is about mutable defaults, which we would eliminate by this and thus avoid potential bugs.

**Review by stefan6419846 — APPROVED** [review]

File: pypdf/constants.py [hunk 2]
```diff
@@ -140,6 +145,62 @@ class Ressources:
...
 
+class Ressources:  # deprecated
+    """
+    Use :class: `Resources` instead.
+
+    .. deprecated:: 5.0.0
+    """
+
+    @classmethod  # type: ignore
+    @property
```
Comment (ztravis) [inline pypdf/constants.py:156]: This only works for python 3.9 and 3.10...

**Pull request merged**